SNIPS/snip-86.md

```diff
@@ -0,0 +1,601 @@
+---
+snip: 86
+title: Sign-In with Starknet (SIWS) Standard
+author: Yudhishthra Sugumaran <yudhishthra.sugumaran@nethermind.io>
+discussions-to: https://community.starknet.io/t/sign-in-with-starknet-technical-proposal/95683/9
+status: Draft
+type: Standards Track
+category: SRC
+created: 2024-05-23
+requires: SNIP-12
+---
+
+## Summary
+
+The Sign-in with Starknet Standard (SIWS) enhances the flexibility and security of user authentication across Starknet by leveraging account abstraction and extended data fields. This standard supports a wider range of authentication scenarios, thereby improving user experience and expanding application development possibilities on Starknet.
+
+## Abstract
+
+The proposed SIWS standard introduces an advanced framework for authentication that enables signing of messages of any length, removing character limitations. This enhancement is made possible through the integration of the Poseidon hash function, which supports larger data sizes. The standard aims to facilitate broader adoption across decentralized applications by offering enhanced flexibility and robust security features.
```

Comment:
Could you kindly point me out where you saw that:
> the Poseidon hash function, which supports larger data sizes.
?

Reply:
Perhaps I can reword it, but what I understood about how the Poseidon hash works is that it has significantly fewer constraints per message bit than Pedersen hash therefore making it suitable to sign larger **message** sizes

Also, would like to highlight on SNIP-12 (https://github.com/starknet-io/SNIPs/blob/main/SNIPS/snip-12.md) which demonstrates this. I also emphasised on sticking to revision 1, cause of this too -> https://github.com/starknet-io/SNIPs/blob/main/SNIPS/snip-12.md?plain=1#L196

Reply:
Thing is that Poseidon was pushed quite a lot due to the fact that it is/will be "more efficient" but apparently with STWO (coming early 2025) Pedersen might be the most efficient one in the end... 
This is an annoying situation 😞 

Yes it means that there might be a new version for SNIP-12 to use the most efficient hashing function.

Reply:
Yikes...I remember the SNIP also mentioning that support for strings longer than 31 characters would be introduced in revision `1` here -> https://github.com/starknet-io/SNIPs/blob/main/SNIPS/snip-12.md?plain=1#L353

Though it seems like this is already supported in Starknet.js after version `6.8.0`